src/components/BurnWarning/BurnWarning.tsx

```diff
@@ -0,0 +1,164 @@
+import { Button, Grid, Popover, Typography } from '@material-ui/core'
+import { BN } from '@project-serum/anchor'
+import useStyles from './style'
+import React, { useEffect } from 'react'
+import { displayDate, divUpNumber, printBN } from '@consts/utils'
+import { Decimal } from '@synthetify/sdk/lib/exchange'
+import { IRewardsProps, RoundData } from '@components/WrappedActionMenu/RewardsTab/RewardsTab'
+import { blurContent, unblurContent } from '@consts/uiUtils'
+
+interface IProps {
+  open: boolean
+  burnAmount: {
+    amount: BN
+    decimal: number
+  }
+  onBurn: (amount: BN, decimals: number) => () => void
+  onCancel: () => void
+  burnTokenSymbol: string
+  rewards: IRewardsProps
+}
+
+const BurnWarning: React.FC<IProps> = ({
+  burnAmount,
+  burnTokenSymbol,
+  onBurn,
+  onCancel,
+  open,
+  rewards
+}) => {
+  const classes = useStyles()
+  useEffect(() => {
+    if (open) {
+      blurContent()
+    } else {
+      unblurContent()
+    }
+  }, [open])
+  const { userDebtShares, roundLength, rounds, slot, amountToClaim } = rewards
+  const estimateRounds = (): RoundData => {
```

Comment:
are these functions from rewards? maybe move these to utils instead of copying

Reply:
done